src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -1378,7 +1381,10 @@ equalityExpression
 
 // boolean relational expressions (level 5)
 relationalExpression
-    :    shiftExpression ( "instanceof"^ typeSpec[true])?
+    :   shiftExpression ( "instanceof"^  (
+            (   typeSpec[true] IDENT)=> patternDefinition
+            |   instanceofTypeSpec  )
+            )?
```

Comment:
Its not easy to follow identing. 1387 looks like it ends 1385 since they have the same indentation but that is not the case. It ends 1384.

Reply:
Fixed, I think.  I always struggle to make the ANTLR grammar look consistent.